lib/utils/cli.go

```diff
@@ -154,12 +154,14 @@ func UserMessageFromError(err error) string {
 	if err != nil {
 		// If the error is a trace error, check if it has a user message embedded in
 		// it, if it does, print it, otherwise escape and print the original error.
-		if er, ok := err.(*trace.TraceErr); ok {
-			if er.Message != "" {
-				return fmt.Sprintf("error: %v", EscapeControl(er.Message))
-			}
+		if err, ok := err.(*trace.TraceErr); ok && err.Message != "" {
+			// Avoid escaping an error message that terminates with a newline
+			errMsg := strings.TrimSuffix(err.Message, "\n")
+			return fmt.Sprintf("error: %v", EscapeControl(errMsg))
 		}
-		return fmt.Sprintf("error: %v", EscapeControl(err.Error()))
+		// Avoid escaping an error message that terminates with a newline
```

Comment:
Interestingly, https://github.com/gravitational/teleport/issues/3205 may be suffering from a similar sort of formatting issue.

Reply:
Indeed, thinking about this, it does not look like the right approach though (in the context of #3205) - it would be better to offload this to a formatting implementation that has an alternative implementation in certain cases so this happens automatically and `tsh` does not need to take any special cases into account.
I'll give more thought.

Reply:
I think I'll go with my original changes and also propagate changes into the forwarding SSH server so it [behaves](https://github.com/gravitational/teleport/pull/4651/commits/0b24af342c484ce0dca173862924ad44313f6588) the same. The last commit also splits the `UserMessageFromError` in two implementation and uses the alternative formatting API to fix https://github.com/gravitational/teleport/issues/3205.